geopandas/geodataframe.py

```diff
@@ -499,7 +506,28 @@ def to_json(self, na="null", show_bbox=False, **kwargs):
           feature individually so that features may have different properties.
         - ``keep``: output the missing entries as NaN.
         """
-        return json.dumps(self._to_geo(na=na, show_bbox=show_bbox), **kwargs)
+        if to_wgs84 is None:
+            if self.crs and not self.crs.equals == 4326:
```

Comment:
```suggestion
            if self.crs and not self.crs.equals(4326):
```

@jorisvandenbossche I think you wanted to do this.

Reply:
Whoops, yes, I was actually trying out `self.crs.equals(4326, ignore_axis_order=True)`, but reverted (incompletely) because this method is only available from pyproj 2.5

Now, the reason I was doing that, is because I realized that EPSG:4326 is not the only valid CRS for which we should not warn. It might be a corner case, but people might have `pyproj.CRS("OGC:CRS84")`, which doesn't equals to 4326 (while this is actually the CRS the geojson spec uses).

Reply:
I've used `self.crs.equals(4326, ignore_axis_order=True)` again, assuming pyproj will be bumped to never version before 0.11 release.

Reply:
I've removed the check from here because it is done in `to_crs` on an array level. No need to duplicate.